operator/v1/types_csi_cluster_driver.go

```diff
@@ -168,6 +168,64 @@ type AWSCSIDriverConfigSpec struct {
 	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov|aws-iso|aws-iso-b|aws-iso-e|aws-iso-f):kms:[a-z0-9-]+:[0-9]{12}:(key|alias)\/.*$`
 	// +optional
 	KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+
+	// efsVolumeMetrics sets the configuration for collecting metrics from EFS volumes used by the EFS CSI Driver.
+	// +openshift:enable:FeatureGate=AWSEFSDriverVolumeMetrics
+	// +optional
+	EFSVolumeMetrics *AWSEFSVolumeMetrics `json:"efsVolumeMetrics,omitempty"`
+}
+
+// AWSEFSVolumeMetricsState defines the modes for collecting volume metrics in the AWS EFS CSI Driver.
+// This can either enable recursive collection of volume metrics or disable metric collection entirely.
+// +kubebuilder:validation:Enum:="RecursiveWalk";"Disabled"
+type AWSEFSVolumeMetricsState string
+
+const (
+	// AWSEFSVolumeMetricsRecursiveWalk indicates that volume metrics collection in the AWS EFS CSI Driver
+	// is performed by recursively walking through the files in the volume.
+	AWSEFSVolumeMetricsRecursiveWalk AWSEFSVolumeMetricsState = "RecursiveWalk"
+
+	// AWSEFSVolumeMetricsDisabled indicates that volume metrics collection in the AWS EFS CSI Driver is disabled.
+	AWSEFSVolumeMetricsDisabled AWSEFSVolumeMetricsState = "Disabled"
+)
+
+// AWSEFSVolumeMetrics defines the configuration for volume metrics in the EFS CSI Driver.
+// +union
+type AWSEFSVolumeMetrics struct {
+	// state defines the state of metric collection in the AWS EFS CSI Driver.
+	// This field is required and must be set to one of the following values:
+	// - Disabled: No metrics collection will be performed. This is the default value.
+	// - RecursiveWalk: The AWS EFS CSI Driver will recursively scan volumes to collect metrics.
+	//   This process may result in high CPU and memory usage, depending on the volume size.
+	// +unionDiscriminator
+	// +kubebuilder:default:=Disabled
+	// +default="Disabled"
+	// +optional
```

Comment:
This field should be required and not have a default, the field referencing the struct is a pointer and optional so this won't cause problems for json marshalling

Reply:
Removed the default and optional; and added required.

About required, do I need `+required` as well? I looked in the code base for references, but I see both ways.

Also, since the field is now required, I suppose I should remove `omitempty`, correct? Again, I looked at other APIs and I see mixed usages, so I'm not sure.

Reply:
`+kubebuilder:validation:Required` and no `omitempty` is preferred, don't need `+required`

Reply:
Perfect, that's what I did.